lib/utils/sudo.ts

```diff
@@ -14,34 +14,78 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { spawn, StdioOptions } from 'child_process';
 
 import * as Bluebird from 'bluebird';
-import * as rindle from 'rindle';
+import { ChildProcess, spawn, SpawnOptions } from 'child_process';
+import { stripIndent } from 'common-tags';
+import * as _ from 'lodash';
 
+/**
+ * Execute a child process with admin / superuser privileges, prompting
+ * the user for elevation as needed.
+ * @param command Path to an executable, plus any arguments.
+ * These will be shell-escaped by this function.
+ * @param stderr Optional stream to which stderr should be piped
+ */
 export async function executeWithPrivileges(
 	command: string[],
 	stderr?: NodeJS.WritableStream,
-): Promise<string> {
-	const stdio: StdioOptions = [
-		'inherit',
-		'inherit',
-		stderr ? 'pipe' : 'inherit',
-	];
-	const opts = {
-		env: process.env,
-		stdio,
-	};
+): Promise<void> {
+	// whether the CLI is already running with admin / super user privileges
+	const isElevated = await (await import('is-elevated'))();
+	const { shellEscape } = await import('./helpers');
+	const escapedArgs = shellEscape(command);
 
-	const args = process.argv
-		.slice(0, 2)
-		.concat(['internal', 'sudo', command.join(' ')]);
```

Comment:
The previous code would call "`internal sudo`" here, which in turn would call `spawn` or `windosu`. It seemed to me like an unnecessary additional child process (executing the CLI itself as a child, and the CLI is unfortunately slow-ish), so I moved the implementation of "`internal sudo`" here (with modifications) and called it directly.